Sources/TootSDK/Models/TootFeature.swift

```diff
@@ -1,8 +1,139 @@
 import Foundation
+import Version
 
 /// Represents a feature that is not supported by all flavours.
-public struct TootFeature {
+public struct TootFeature: Equatable {
 
-    /// Flavours that support this feature.
-    public let supportedFlavours: Set<TootSDKFlavour>
+    /// Represents a flavour requirement with optional version constraints
+    public struct FlavourRequirement: Equatable {
+        let flavour: TootSDKFlavour
+        let minVersion: Version?
+        let maxVersion: Version?
+
+        /// Create a requirement for any version of a flavour
+        public static func any(_ flavour: TootSDKFlavour) -> FlavourRequirement {
+            FlavourRequirement(flavour: flavour, minVersion: nil, maxVersion: nil)
+        }
+
+        /// Create a requirement for a minimum version
+        public static func from(_ flavour: TootSDKFlavour, version: String) -> FlavourRequirement {
+            FlavourRequirement(flavour: flavour, minVersion: Version(tolerant: version), maxVersion: nil)
+        }
+
+        /// Create a requirement for a version range
+        public static func range(_ flavour: TootSDKFlavour, from: String, to: String) -> FlavourRequirement {
+            FlavourRequirement(
+                flavour: flavour,
+                minVersion: Version(tolerant: from),
+                maxVersion: Version(tolerant: to)
+            )
+        }
+
+        /// Create a requirement for a maximum version
+        public static func until(_ flavour: TootSDKFlavour, version: String) -> FlavourRequirement {
+            FlavourRequirement(flavour: flavour, minVersion: nil, maxVersion: Version(tolerant: version))
+        }
+    }
+
+    /// The requirements for this feature
+    public let requirements: [FlavourRequirement]
+
+    /// Legacy initializer for backward compatibility
+    public init(supportedFlavours: Set<TootSDKFlavour>) {
+        self.requirements = supportedFlavours.map { FlavourRequirement.any($0) }
+    }
+
+    /// New version-aware initializer
+    public init(requirements: [FlavourRequirement]) {
+        self.requirements = requirements
+    }
+
+    /// Initialize with "any flavour except specific version requirements" pattern
+    /// - Parameters:
+    ///   - anyExcept: Flavours that should support any version
+    ///   - versionRequirements: Specific version requirements for certain flavours
+    public init(anyExcept: Set<TootSDKFlavour>, versionRequirements: [FlavourRequirement]) {
```

Comment:
This init is unclear to me. It reads as: support any flavor except these in `anyExcept`. I suggest `anyVersion` and `requirements` as parameter names.